src/app/screens/Register/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import i18next from 'i18next';
+import { useForm } from 'react-hook-form';
+
+import AuthWrapper from '~components/AuthWrapper';
+import Input from '~components/Input';
+import { emailRegex } from '~utils/inputValidations';
+import { AUTH_FIELDS } from '~constants/fields';
+
+import styles from './styles.module.scss';
+
+interface DataForm {
+  email: string;
+  firstName: string;
+  lastName: string;
+  locale: string;
+  password: string;
+  passwordConfirmation: string;
+}
+
+function Register() {
+  const { register, handleSubmit, errors, formState, getValues } = useForm({
+    mode: 'all'
+  });
+
+  const submitForm = (data: DataForm) => {
+    data.locale = i18next.language;
+    const user = {
+      user: data
+    };
+    console.log(user);
+  };
+
+  const validatePassword = (passwordConfirmation: string) =>
+    passwordConfirmation === getValues('password') ? true : 'FormValidations:passwordNotMatch';
```

Comment:
sorry if i made you change it, but its better to use `watch` here because in the other side, i think that the validations only runs when you submit